lib/cylc/time_parser.py

```diff
@@ -223,13 +223,18 @@ def parse_recurrence(self, expression,
             # Convert the exclusion strings to ISO8601 points
             if exclusions is not None:
                 for exclusion in exclusions:
-                    exclusion_point, excl_off = (
-                        self._get_point_from_expression(
-                            exclusion, None, is_required=False,
-                            allow_truncated=False))
-                    if excl_off:
-                        exclusion_point += excl_off
-                    exclusion_points.append(exclusion_point)
+                    try:
+                        # Attempt to convert to TimePoint
+                        exclusion_point, excl_off = (
+                            self._get_point_from_expression(
+                                exclusion, None, is_required=False,
+                                allow_truncated=False))
+                        if excl_off:
+                            exclusion_point += excl_off
+                        exclusion_points.append(exclusion_point)
+                    except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```